src/components/worker/worker.py

```diff
@@ -109,65 +109,204 @@ def __init__(
+                sync = self.pipeline.create(dai.node.Sync)
+                xOut = self.pipeline.create(dai.node.XLinkOut)
+
+                xOut.setStreamName(self.pc_stream_name)
+                xOut.input.setBlocking(False)
+
+                # Link the nodes
+                depth_node.depth.link(pointcloud.inputDepth)
+                if color_node:
+                    color_node.isp.link(sync.inputs["rgb"])
+                pointcloud.outputPointCloud.link(sync.inputs["pcl"])
+                sync.out.link(xOut.input)
+
+        self.pipeline = dai.Pipeline()
+        try:
+            stage = "color"
+            color_node = configure_color()
+
+            stage = "stereo"
+            depth_node = configure_stereo()
 
-        if (
-            not self.should_exec
-        ):  # possible SIGTERM between initializing OakCamera and here
-            LOGGER.debug("Stopping configuration due to termination signal")
-            return
+            stage = "point cloud"
+            configure_pc(color_node, depth_node)
 
-        self._config_oak_camera()
+            LOGGER.info("Successfully configured pipeline.")
+        except Exception as e:
+            msg = f"Error configuring pipeline at stage '{stage}': {e}"
+            resolution_err_substr = "bigger than maximum at current sensor resolution"
+            calibration_err_substr = "no Camera data available"
+            if resolution_err_substr in str(e):
+                msg += ". Please adjust 'height_px' and 'width_px' in your config to an accepted resolution."
+            elif calibration_err_substr in str(e):
+                msg += ". If using a non-integrated model, please check that the camera is calibrated properly."
+            LOGGER.error(msg)
 
         self.message_synchronizer = MessageSynchronizer()
         self.configured = True
 
-    def start(self):
-        if self.oak:
-            self.oak.start()
-            LOGGER.info("Started OakCamera")
-        else:
-            raise AttributeError(
-                "oak.start() called before oak was assigned. Must configure worker first."
+    async def start(self):
+        self.device = None
+        while not self.device and self.should_exec:
+            try:
+                self.device = dai.Device(self.pipeline)
+                LOGGER.debug("Successfully initialized device.")
+            except Exception as e:
+                LOGGER.error(f"Error initializing device: {e}")
+                await asyncio.sleep(1)
+
+        self.color_sensor_queues: List[SensorAndQueue] = []
+        for cs in self.cfg.sensors.color_sensors:
+            q = self.device.getOutputQueue(cs.get_unique_name(), 30, blocking=False)
+            self.color_sensor_queues.append(SensorAndQueue(cs, q))
+
+        if self.cfg.sensors.stereo_pair:
+            self.depth_queue = self.device.getOutputQueue(
+                self.depth_stream_name, 30, blocking=False
```

Comment:
[q] Is it ok to have hardcoded queue size here?

Reply:
Done (unhardcoded it and cleaned up some defaults at the top of the file)